### PR TITLE
[feature] (#324): Support JSON genesis block for integration tests

### PR DESCRIPTION
**Pull request by arndey**

### CODE REVIEW

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaContainer.kt [hunk 1]
```diff
@@ -116,7 +119,7 @@ open class IrohaContainer : GenericContainer<IrohaContainer> {
     override fun start() {
         logger().debug("Starting Iroha container")
         if (logger().isDebugEnabled) {
-            val genesisAsJson = config.genesis.asJson()
+            val genesisAsJson = config.genesis?.asJson()
```
Comment (Mingela) [inline modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaContainer.kt:122]: actually it's going to use `Genesis.getEmpty()` right? 
should be smth like `<genesisPath's value> ?: config.genesis?.asJson()`

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaConfig.kt [hunk 2]
```diff
@@ -24,17 +24,15 @@ import java.util.function.Consumer
 class IrohaConfig(
     var networkToJoin: Network = newNetwork(),
     var logConsumer: Consumer<OutputFrame> = Slf4jLogConsumer(getLogger(IrohaContainer::class.java)),
-    var genesis: Genesis = Genesis.getEmpty(),
+    var genesisPath: String? = null, // first option
+    var genesis: Genesis? = Genesis.getEmpty(), // second option
```
Comment (Mingela) [inline modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaConfig.kt:28]: why nullable if default value is going to always be here

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt [hunk 3]
```diff
@@ -136,24 +136,30 @@ class IrohaRunnerExtension : InvocationInterceptor, BeforeEachCallback {
...
                 val container = IrohaContainer {
                     networkToJoin = network
-                    genesis = withIroha.sources.map { it.createInstance() }.toSingle()
+                    genesisPath = when {
+                        withIroha.source.isEmpty() -> null
+                        else -> withIroha.source
+                    }
+                    genesis = when {
+                        withIroha.source.isEmpty() -> withIroha.sources.map { it.createInstance() }.toSingle()
+                        else -> null
```
Comment (Mingela) [inline modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt:161]: should be empty genesis instead I guess

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt [hunk 4]
```diff
@@ -163,8 +169,7 @@ class IrohaRunnerExtension : InvocationInterceptor, BeforeEachCallback {
                             it.first() to it.last()
                         }
                     }
-                    // only first peer should have --submit-genesis in peer start command
-                    submitGenesis = n == 0
+                    submitGenesis = n == 0 // only first peer should have --submit-genesis in peer start command
```
Comment (Mingela) [inline modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt:172]: I'd prefer previous formatting of the comment

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt [hunk 5]
```diff
@@ -136,24 +136,30 @@ class IrohaRunnerExtension : InvocationInterceptor, BeforeEachCallback {
...
             async(Dispatchers.IO) {
                 val p2pPort = portsList[n][IrohaConfig.P2P_PORT_IDX]
                 val container = IrohaContainer {
                     networkToJoin = network
-                    genesis = withIroha.sources.map { it.createInstance() }.toSingle()
+                    genesisPath = when {
+                        withIroha.source.isEmpty() -> null
+                        else -> withIroha.source
+                    }
+                    genesis = when {
```
Comment (Mingela) [inline modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt:159]: how about actually creating a genesis just here and get rid of the `genesisPath` in `IrohaConfig`?

**Review by Mingela — COMMENTED** [review]
consider comments/questions from above

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaConfig.kt (hunk 2 above)
Comment (Mingela) [reply modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaConfig.kt:28]: then why not `null` as a default value and throw an exception if nothing is set, or there is a point in running Iroha with empty genesis?

**Review by gv-timur — APPROVED** [review]

**Review by Mingela — APPROVED** [review]

**Pull request merged**